skbuild/constants.py

```diff
@@ -20,10 +20,15 @@ def _default_skbuild_plat_name():
     On macOS, it corresponds to the version and machine associated with :func:`platform.mac_ver()`.
     """
     if sys.platform == 'darwin':
-        # distutils.util.get_platform() returns the release based on the value
-        # of MACOSX_DEPLOYMENT_TARGET on which Python was built, which may
-        # be significantly older than the user's current machine.
+        # If the MACOSX_DEPLOYMENT_TARGET environment variable is defined, use
```

Comment:
For the comment, I instead suggest something like this:
```
        # If the MACOSX_DEPLOYMENT_TARGET environment variable is defined, use
        # it, as it will be the most accurate. Otherwise use the value returned
        # by platform module from the Python standard library.
        # 
        # Note that since distutils.util.get_platform() returns the release based
        # on the value of MACOSX_DEPLOYMENT_TARGET on which Python was built,
        # (which may be significantly older than the user's current machine), it can not be used.
```

Reply:
The very last sentence is confusing. Instead of

> (which may be significantly older than the user's current machine), it can not be used.

I would suggest

> (which may be significantly older than the user's current machine), it should only be used if the MACOSX_DEPLOYMENT_TARGET environment variable is not defined.

Reply:
In fact there are two ways of getting the platforms:

```
import platform
print("approach 1", platform.mac_ver())

from distutils.util import get_platform
print("approach 2", get_platform())
```

On macOS, the approach1 is used and the comment `Note that since distutils.util.get_platform() ...` is here to explain why approach 2 is not used.

Reply:
That assumes that `MACOSX_DEPLOYMENT_TARGET` is always defined on macOS. I don't know if that's always necessarily the case. So the code should still work on macOS even if MACOSX_DEPLOYMENT_TARGET is not defined. 

I didn't know about [platform.mac_ver()](https://docs.python.org/3.8/library/platform.html#mac-os-platform). Should we use that instead of the `MACOSX_DEPLOYMENT_TARGET` environment variable?

Reply:
> I didn't know about platform.mac_ver().  Should we use that instead of the MACOSX_DEPLOYMENT_TARGET environment variable?

It is already used to get the target. Your proposed patch makes sense.

> That assumes that MACOSX_DEPLOYMENT_TARGET is always defined on macOS.

You are right, MACOSX_DEPLOYMENT_TARGET is not always defined.


Here is an updated comment:

```
        # If the MACOSX_DEPLOYMENT_TARGET environment variable is defined, use
        # it, as it will be the most accurate. Otherwise use the value returned
        # by platform.mac_ver() provided by the platform module available in the Python standard library.
        # 
        # Note that on macOS, distutils.util.get_platform() is not used because it returns
        # the macOS version on which Python was built which may be significantly older than the user's 
        # current machine.
```

Reply:
Ah, my bad, for some reason I thought the `distutils.util.get_platform()` was used. 

I am still confused though: one would think the `platform.mac_ver()` would give the correct number, and yet we found that the number in `MACOSX_DEPLOYMENT_TARGET` was correct. How is that possible?

Reply:
Comment updated in d92bce27dd4dd12c4675e346f5b21c40555ea4b2.